bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmBridgeHandler.java

```diff
@@ -0,0 +1,645 @@
+                } catch (NumberFormatException e) {
+                    logger.warn("Bridge: Badly formatted version number {}", softwareVersion);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Badly formatted version number");
+                }
+            } else {
+                logger.warn("Bridge: Can't parse software version {}", softwareVersion);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Can't parse software version");
+            }
+        } else {
+            logger.warn("Bridge: Unable to determine Zoneminder software version");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Software version is null");
+        }
+        return false;
+    }
+
+    private boolean checkApiVersion(@Nullable String apiVersion) {
+        logger.debug("Bridge: Zoneminder API version is {}", apiVersion);
+        if (apiVersion != null) {
+            String[] versionParts = apiVersion.split("\\.");
+            if (versionParts.length >= 2) {
+                try {
+                    int versionMajor = Integer.parseInt(versionParts[0]);
+                    if (versionMajor >= 2) {
+                        logger.debug("Bridge: Zoneminder API version check OK");
+                        return true;
+                    } else {
+                        logger.warn("Bridge: Requires API version >= 2.0. This Zoneminder is version {}", apiVersion);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Requires API version >= 2.0");
+                    }
+                } catch (NumberFormatException e) {
+                    logger.warn("Bridge: Badly formatted version number {}", apiVersion);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Badly formatted API version number");
+                }
+            } else {
+                logger.warn("Bridge: Can't parse software version {}", apiVersion);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Can't parse API version");
+            }
+        } else {
+            logger.warn("Bridge: Unable to determine Zoneminder API version");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "API version is null");
+        }
+        return false;
+    }
+
+    /*
+     * The refresh job is executed every second
+     * - updates the monitor handlers every monitorsInterval seconds, and
+     * - runs the monitor discovery every discoveryInterval seconds
+     */
+    private void refresh() {
+        try {
+            refreshMonitors();
+            discoverMonitors();
+        } catch (RuntimeException e) {
+            logger.info("Bridge: Unexpected exception during refresh", e);
```

Comment:
Logging to info should be used rarely. This could be warn.

Reply:
Agreed. However, in this case since the exception is not expected, and may not be easily reproduced, I would prefer it be logged when it occurs. Ok?

Reply:
I'll make it warn.